wordpress/wp-content/plugins/cds-base/classes/Modules/Forms/Subscribe/Unsubscribe.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CDS\Modules\Forms\Subscribe;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\ServerException;
+use WP_REST_Request;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\UriInterface;
+
+class Unsubscribe
+{
+    protected string $redirect = '';
+
+    public function unsubscribe(WP_REST_Request $request)
+    {
+        $subscriptionId = $request['id'];
+
+        $client = new Client();
+
+        $endpoint = getenv('LIST_MANAGER_ENDPOINT');
+
+        try {
+            $client->request('GET', $endpoint . '/unsubscribe/' . $subscriptionId, [
+                'allow_redirects' => [
+                    'on_redirect' => function (
+                        RequestInterface $request,
+                        ResponseInterface $response,
+                        UriInterface $uri
+                    ) {
+                        $this->redirect = strval($uri);
+                    }
+                ]
+            ]);
+        } catch (ClientException $e) {
+            error_log($e->getMessage());
+            wp_redirect('/list-manager-error');
+            exit();
+        } catch (ServerException $e) {
+            error_log($e->getMessage());
```

Comment:
Might want to add a partial string we can search the logs for here `Unsubscribe - `